### PR TITLE
Discovery Service with AWS instance 

### PR DESCRIPTION
**Pull request by nagulmeeras**

### CODE REVIEW

File: src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala [hunk 1]
```diff
@@ -69,24 +69,22 @@ object APMServerDetails {
...
-                  case `apmServer_site_relation` =>
-                    enityMap.put("siteEntity", Site.fromNeo4jGraph(relationship.getEndNode.getId))
 
-                  case `apmServer_header_relation` =>
-                    enityMap.put("headers", Some(relationship.getEndNode.getAllProperties.foldLeft(Map[String, String]())((map, property) => map + ((property._1, property._2.asInstanceOf[String])))))
+            val tupleObj = node.getRelationships.foldLeft(Tuple2[AnyRef, Map[String, String]](AnyRef, Map.empty[String, String])) {
+              (tuple, relationsip) =>
+                val childNode = relationsip.getEndNode
+                relationsip.getType.name match {
+                  case `apmServer_site_relation` => Tuple2(Site.fromNeo4jGraph(childNode.getId), tuple._2)
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala]: You don't actually have to call Tuple2 you ca just write (a,b). Same goes for type signature as well.

File: src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+object AWSComputeAPI {
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  def getInstances(amazonEC2: AmazonEC2 , accountInfo: AccountInfo): List[Instance] = {
+
+      val awsInstancesResult = getAWSInstances(amazonEC2)
+      val totalsecurityGroups = amazonEC2.describeSecurityGroups.getSecurityGroups.foldLeft(Map[String, SecurityGroup]())((map, sg) => map + ((sg.getGroupId, sg)))
+      val addresses = amazonEC2.describeAddresses.getAddresses.foldLeft(Map[String, Address]())((map, address) => map + ((address.getInstanceId, address)))
+      val imageIds = awsInstancesResult.foldLeft(List[String]())((list, awsInstance) => list.::(awsInstance.getImageId))
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala]: Why list.:: ?

File: src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala (hunk 1 above)
Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala]: Usage of explicit Tuple() signature is removed

File: src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala (hunk 2 above)
Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala]: for adding imageId of each AWSInstance to List.
Should i change it like this  list++List(valueToBeAdded) ?

Comment (eklavya) [reply src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala]: I mean :: without the preceding dot.

File: src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala (hunk 1 above)
Comment (eklavya) [reply src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala]: removed where?

File: src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      neo =>
+        val node = repository.getNodeById(nodeId)(neo)
+        if (repository.hasLabel(node, snapshotInfoLabel)) {
+          val keyValueInfo = node.getRelationships.foldLeft(List.empty[AnyRef]) {
+            (list, relationship) =>
+              list.::(KeyValueInfo.fromNeo4jGraph(relationship.getEndNode.getId))
+          }
+          Some(SnapshotInfo(
+            Some(nodeId),
+            repository.getProperty[String](node, "snapshotId").get,
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala]: all these .get is it impossible to get a None here?

File: src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala (hunk 2 above)
Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/AWSComputeAPI.scala]: Removed .:: and used :: in place now

File: src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala (hunk 3 above)
Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/SnapshotInfo.scala]: Handled the case if the node is not present

File: src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala (hunk 1 above)
Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala]: Removed it and pushed the code now

**Pull request merged**